prefect_kubernetes/jobs.py

```diff
@@ -299,3 +309,204 @@ def kubernetes_orchestrator():
+                        "Kubernetes pod logs for more information."
+                    )
+                elif latest_v1_job.status.succeeded:
+                    self._completed = True
+                    self.logger.info(
+                        f"Job {latest_v1_job.metadata.name!r} has completed."
+                    )
+
+        if self._kubernetes_job.delete_after_completion:
+            await self._cleanup()
+
+    async def fetch_result(self) -> Dict[str, Any]:
+        """Fetch the results of the job.
+
+        Returns:
+            The logs from each of the pods in the job.
+
+        Raises:
+            ValueError: If this method is called when the job has
+                a non-terminal state.
+        """
+
+        if not self._completed:
+            raise ValueError(
+                "The Kubernetes Job run is not in a completed state - "
+                "be sure to call `wait_for_completion` before attempting "
+                "to fetch the result."
+            )
+        return self.pod_logs
+
+
+class KubernetesJob(JobBlock):
+    """A block representing a Kubernetes job configuration."""
+
+    v1_job: Dict[str, Any] = Field(
+        default=...,
+        description=(
+            "The Kubernetes job manifest to run. This dictionary can be produced "
+            "using `yaml.safe_load`."
+        ),
+    )
+    api_kwargs: Dict[str, Any] = Field(
+        default_factory=dict,
+        description="The kwargs to pass to all Kubernetes API calls.",
+        example={"pretty": "true"},
+    )
+    credentials: KubernetesCredentials = Field(
+        default=..., description="The credentials to configure a client from."
+    )
+    delete_after_completion: bool = Field(
+        default=True,
+        description="Whether to delete the job after it has completed.",
+    )
+    interval_seconds: int = Field(
+        default=5,
+        description="The number of seconds to wait between job status checks.",
+    )
+    namespace: str = Field(
+        default="default",
+        description="The namespace to create and run the job in.",
+    )
+    timeout_seconds: Optional[int] = Field(
+        default=None,
+        description="The number of seconds to wait for the job run before timing out.",
+    )
+
+    _block_type_name = "Kubernetes Job"
+    _logo_url = "https://images.ctfassets.net/zscdif0zqppk/531JKlIwMeEXcBnoK0yeB8/e304dc11a9d25c831901e9cd668433fa/Kubernetes_logo_without_workmark.svg.png?h=250"  # noqa: E501
+
+    async def trigger(self):
+        """Create a Kubernetes job and return a `KubernetesJobRun` object.
+
+        While altering `self` here is not ideal, it avoids the
+        need to use JSON encoders for the UI, define a `dict` method,
+        and bypass the validation caused by using
+        `convert_manifest_to_model` in `block_initialization`.
+        """
+
+        self.v1_job = convert_manifest_to_model(self.v1_job, "V1Job")
```

Comment:
I'm not sure if we should override a user provided variable like this; maybe we can use `self._v1_job_model`?

Reply:
yeah that's probably a good idea - just did that